ceph-aio-no-vagrant.sh

```diff
@@ -95,37 +114,57 @@ function cp_var {
 }
 
 function populate_vars {
-  sed -i "s/#osd_auto_discovery: false/osd_auto_discovery: true/" group_vars/osds.yml
-  sed -i "s/#journal_collocation: false/journal_collocation: true/" group_vars/osds.yml
-  sed -i "s/#pool_default_size: 3/pool_default_size: 2/" group_vars/all.yml
-  sed -i "s/#monitor_address: 0.0.0.0/monitor_address: ${IP}/" group_vars/all.yml
-  sed -i "s/#journal_size: 0/journal_size: 100/" group_vars/all.yml
-  sed -i "s|#public_network: 0.0.0.0\/0|public_network: ${SUBNET}|" group_vars/all.yml
-  sed -i "s/#common_single_host_mode: true/common_single_host_mode: true/" group_vars/all.yml
+  sed -i "s/[#]*osd_auto_discovery: .*/osd_auto_discovery: true/" group_vars/osds.yml
+  sed -i "s/[#]*journal_collocation: .*/journal_collocation: true/" group_vars/osds.yml
+  sed -i "s/[#]*monitor_address: .*/monitor_address: ${IP}/" group_vars/all.yml
+  sed -i "s/[#]*journal_size: .*/journal_size: 100/" group_vars/all.yml
+  sed -i "s|[#]*public_network: .*|public_network: ${SUBNET}|" group_vars/all.yml
+  sed -i "s/[#]*common_single_host_mode: .*/common_single_host_mode: true/" group_vars/all.yml
+
+  if ! grep -q '^ceph_conf_overrides:' group_vars/all.yml; then
+    cat >> group_vars/all.yml <<EOF
+ceph_conf_overrides:
+  global:
+    mon pg warn max per osd: 0
```

Comment:
Spaces won't work here so you need to:

s/mon pg warn max per osd/mon_pg_warn_max_per_osd/